src/linear.cpp

```diff
@@ -74,22 +75,18 @@ struct linearWorker : public Worker {
                 v1 = rowvec(trans(mt1t * mt2.col(i)));
                 v2 = center1 * center2[i] * ncol;
                 simils = to_vector(((v1 - v2) / ncol) / (square1 * square2[i]));
-                for (std::size_t j = 0; j < square1.size(); j++) {
-                    if (square1[j] == 0.0 || square2[i] == 0.0)
-                        simils[j] = use_nan ? std::numeric_limits<double>::quiet_NaN() : 0.0;
```

Comment:
If you remove this, will you not get the same issue as in #20? Example:

``` r
mat1 <- Matrix::Matrix(1:4, nrow = 1, sparse = TRUE)
mat2 <- Matrix::Matrix(rep(0.1111, 4), nrow = 1, sparse = TRUE)
mat3 <- Matrix::Matrix(rep(1, 4), nrow = 1, sparse = TRUE)
simil(mat1, mat2, method = "correlation")
# [1,] Inf
simil(mat1, mat2, method = "correlation", diag = TRUE)
# [1,] Inf
simil(mat1, mat3, method = "correlation")
# [1,] .
simil(mat1, mat3, method = "correlation", diag = TRUE)
# [1,] 0

Reply:
Thanks. I shouldn't have removed `replace_inf()` (f93fde0). I don't know why the tests did not detect this...